src/components/ExpensiTextInput/BaseExpensiTextInput.js

```diff
@@ -154,23 +167,41 @@ class BaseExpensiTextInput extends Component {
                                     labelScale={this.state.labelScale}
                                 />
                             ) : null}
-                            <TextInput
-                                ref={(ref) => {
-                                    if (typeof innerRef === 'function') { innerRef(ref); }
-                                    this.input = ref;
-                                }}
-                                // eslint-disable-next-line
-                                {...inputProps}
-                                value={value}
-                                placeholder={(this.state.isFocused || !label) ? placeholder : null}
-                                placeholderTextColor={themeColors.placeholderText}
-                                underlineColorAndroid="transparent"
-                                style={inputStyle}
-                                multiline={multiline}
-                                onFocus={this.onFocus}
-                                onBlur={this.onBlur}
-                                onChangeText={this.setValue}
-                            />
+                            <View style={styles.flexRow}>
+                                <TextInput
+                                    ref={(ref) => {
+                                        if (typeof innerRef === 'function') {innerRef(ref);}
+                                        this.input = ref;
+                                    }}
+                                    // eslint-disable-next-line
+                                    {...inputProps}
+                                    value={value}
+                                    placeholder={(this.state.isFocused || !label) ? placeholder : null}
+                                    placeholderTextColor={themeColors.placeholderText}
+                                    underlineColorAndroid="transparent"
+                                    style={[...inputStyle, styles.flex1, styles.pt3, !hasLabel && styles.pv0]}
+                                    onFocus={this.onFocus}
+                                    onBlur={this.onBlur}
+                                    onChangeText={this.setValue}
+                                    secureTextEntry={this.state.passwordHidden}
+                                />
+                                {secureTextEntry && (
```

Comment:
I feel like this part can go into a separate component which only deals with PasswordInput. I would love to hear your thoughts.

Reply:
I considered this.
Do you mean a wrapper secure text component which passes in the icon control via props to this component?

Reply:
So something like called as `PasswordExpensiInput`.

```javascript
<Wrapper>
  <ExpensiInput secureInput/>
  <icon>
</Wrapper>
```

Reply:
But the icon needs to be inside the layout of the input as it needs to take up space inside the flow of the flex container to displace any icons added inside the input by password managers.